timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -23,9 +23,10 @@ NonbondedInteractionGroup<RealType>::NonbondedInteractionGroup(
     const std::vector<int> &row_atom_idxs,
     const double beta,
     const double cutoff,
+    const std::optional<std::set<int>> &col_atom_idxs,
     const bool disable_hilbert_sort,
     const double nblist_padding)
-    : N_(N), NR_(row_atom_idxs.size()), NC_(N_ - NR_),
+    : N_(N), NR_(row_atom_idxs.size()), NC_(col_atom_idxs ? col_atom_idxs->size() : N_ - NR_), K_(NR_ + NC_),
```

Comment:
does `std::optional<std::set<int> >col_atom_idxs` overload the boolean operator so you can evaluate it like this? This is starting to hurt my brain.

Reply:
Yes it does, but removed in the next commit